Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -455,13 +453,10 @@ static gboolean webkitMediaCommonEncryptionDecryptSinkEventHandler(GstBaseTransf
             result = TRUE;
             ASSERT(cdmInstance);
             LockHolder locker(priv->m_mutex);
-            if (priv->m_cdmInstance != cdmInstance) {
-                GST_INFO_OBJECT(self, "got new CDMInstance %p attached (ours was %p), clearing state", cdmInstance, priv->m_cdmInstance.get());
-                webKitMediaCommonEncryptionDecryptorPrivClearStateWithInstance(priv, cdmInstance);
-                if (!priv->m_pendingProtectionEvents.isEmpty())
-                    webkitMediaCommonEncryptionDecryptProcessPendingProtectionEvents(self);
-            } else
-                GST_TRACE_OBJECT(self, "got attach CDMInstance for the same instance %p we already have", cdmInstance);
+            GST_INFO_OBJECT(self, "got new CDMInstance %p attached (ours was %p), clearing state", cdmInstance, priv->m_cdmInstance.get());
+            webKitMediaCommonEncryptionDecryptorPrivClearStateWithInstance(priv, cdmInstance);
+            if (!priv->m_pendingProtectionEvents.isEmpty())
+            webkitMediaCommonEncryptionDecryptProcessPendingProtectionEvents(self);
```

Comment:
Why keep the check if we can have an invariant it will always be a new CDM instance?

Reply:
Because it is not a big deal and it can protect us from changes in the future as it keeps things more decoupled. Anyway, I don't have a strong opinion on this.